lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -1008,6 +1084,203 @@ prop_coinSelectionLens_givesPriorityToCoins (Blind (Small u)) =
+--
+-- We expect no splitting of token bundles.
+--
+boundaryTest2 :: BoundaryTestData
+boundaryTest2 = BoundaryTestData
+    { boundaryTestCriteria = BoundaryTestCriteria {..}
+    , boundaryTestExpectedResult = BoundaryTestResult {..}
+    }
+  where
+    assetA = AssetId (UnsafeTokenPolicyId $ Hash "A") (UnsafeTokenName "1")
+    q1 :| [q2] = equipartitionTokenQuantity maxTxOutTokenQuantity (() :| [()])
+    boundaryTestOutputs =
+      [ (Coin 1_500_000, []) ]
+    boundaryTestUTxO =
+      [ (Coin 1_000_000, [(assetA, q1)])
+      , (Coin 1_000_000, [(assetA, q2)])
+      ]
+    boundaryTestInputs =
+      [ (Coin 1_000_000, [(assetA, q1)])
+      , (Coin 1_000_000, [(assetA, q2)])
+      ]
+    boundaryTestChange =
+      [ (Coin 500_000, [(assetA, maxTxOutTokenQuantity)]) ]
+
+-- Slightly exceed the maximum token quantity by selecting inputs with the
+-- following quantities:
+--
+--  - Quantity #1: 1
+--  - Quantity #2: maximum token quantity
+--
+-- We expect splitting of change bundles.
+--
+boundaryTest3 :: BoundaryTestData
+boundaryTest3 = BoundaryTestData
+    { boundaryTestCriteria = BoundaryTestCriteria {..}
+    , boundaryTestExpectedResult = BoundaryTestResult {..}
+    }
+  where
+    assetA = AssetId (UnsafeTokenPolicyId $ Hash "A") (UnsafeTokenName "1")
+    q1 :| [q2] = equipartitionTokenQuantity
+        (TokenQuantity.succ maxTxOutTokenQuantity) (() :| [()])
+    boundaryTestOutputs =
+      [ (Coin 1_500_000, []) ]
+    boundaryTestUTxO =
+      [ (Coin 1_000_000, [(assetA, TokenQuantity 1)])
+      , (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      ]
+    boundaryTestInputs =
+      [ (Coin 1_000_000, [(assetA, TokenQuantity 1)])
+      , (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      ]
+    boundaryTestChange =
+      [ (Coin 250_000, [(assetA, q1)])
+      , (Coin 250_000, [(assetA, q2)])
+      ]
+
+-- Reach (but do not exceed) exactly twice the maximum token quantity by
+-- selecting inputs with the following quantities:
+--
+--  - Quantity #1: maximum token quantity
+--  - Quantity #2: maximum token quantity
+--
+-- We expect splitting of change bundles.
+--
+boundaryTest4 :: BoundaryTestData
+boundaryTest4 = BoundaryTestData
+    { boundaryTestCriteria = BoundaryTestCriteria {..}
+    , boundaryTestExpectedResult = BoundaryTestResult {..}
+    }
+  where
+    assetA = AssetId (UnsafeTokenPolicyId $ Hash "A") (UnsafeTokenName "1")
+    boundaryTestOutputs =
+      [ (Coin 1_500_000, []) ]
+    boundaryTestUTxO =
+      [ (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      , (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      ]
+    boundaryTestInputs =
+      [ (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      , (Coin 1_000_000, [(assetA, maxTxOutTokenQuantity)])
+      ]
+    boundaryTestChange =
+      [ (Coin 250_000, [(assetA, maxTxOutTokenQuantity)])
+      , (Coin 250_000, [(assetA, maxTxOutTokenQuantity)])
+      ]
+
```

Comment:
It'd have been nice to also show an example where some assets are specified in the outputs, but the resulting change is still going above limit.